File: python/src/logica_programacion/ejercicio1.py
# Resultado ejercicio #1:


def tabla_de_multiplicar(tabla, rango=10):
    """
    Genera la tabla de multiplicar para un número dado hasta el rango especificado.

    :param tabla: int, La tabla de multiplicar a generar.
    :param rango: int (opcional), hasta qué número se debe multiplicar la 'tabla'.
    :return: str, la tabla de multiplicar como un string formateado.
    """

    tabla_multiplicar = f"Tabla del #{tabla} \n"
    for numero in range(0, rango + 1):
        resultado = tabla * numero
        tabla_multiplicar += f'{tabla} x {numero} = {resultado} \n'

    return tabla_multiplicar


# print(tabla_de_multiplicar(2))

File: typescript/src/ejercicio1.ts
// Resultado ejercicio #1:


export let tablaDeMultiplicar = (tabla: number, rango=10) => {
    /**
     * Genera la tabla de multiplicar para un número dado hasta el rango especificado.
     * @param tabla: number, La tabla de multiplicar a generar.
     * @param rango: number (opcional), hasta qué número se debe multiplicar la 'tabla'.
     * @return: string, la tabla de multiplicar como un string formateado.
     */

    let tablaMultiplicar = `Tabla del #${tabla} \n`

    for (let i:number = 0; i <= rango; i++) {
        let resultado = tabla * i
        tablaMultiplicar += `${tabla} x ${i} = ${resultado} \n`
    }

    return tablaMultiplicar
}

// console.log(tablaDeMultiplicar(2))
